### DISCUSSION TITLE
Add the ability to search for field names in the query builder

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5820] [Feature Requests]
<img width="1601" alt="image" src="https://github.com/specify/specify7/assets/37256050/e459e060-0640-4c3d-9373-2228155b5e05">

In this interface, I imagine it would show a list of options similar to how the auto-mapper suggests fields based on column name in the WorkBench.

While many users are familiar with the field name, they often do not know how it is related to their base table. Being able to search for a field name would be extremely useful. I have some concerns about implementing this feature, mainly that there are _many, many_ relationships with duplicate names for fields (for instance, "Full Name" would be shared between all trees) and that may be difficult to visualize. Another concern is that relationships can be much more extensive than people recognize, so we may want to limit how many rels out the query term can go.

@melton-jason likely has similar concerns. @realVinayak knows how bad a large number of joins are– and this search option may make people choose the wrong field without thinking.

@CarolineDenis the example design above is _super_ basic, not exactly the best way to do it so if you implement this I know you would do it better than me 😂

**Requested By:** Geneva and many others

**Comment by maxpatiiuk** [comment]
Related to https://github.com/specify/specify7/issues/1253

**Comment by grantfitzsimmons** [comment]
I swear I saw a duplicate of this (really, this is the duplicate) with much fewer details. Please find it and close it GitHub wizards.

**Comment by maxpatiiuk** [comment]
Your Majesty, Emperor of the Gitlands and Supreme Sorcerer of the Code Realms,

With the swish of my digital wand and the guidance of my mystical binary crystal ball, I have traversed the vast repositories of GitHub to locate the elusive twin issue. Behold, the enigmatic Issue #2274!

Fear not, for I shall command the mighty powers of 'Duplicate' and 'Close' to quell this rebellion in the codebase. The realms of Redlands and Lawrence shall once again bask in the harmony of well-organized issues.

Yours in ludicrous code wizardry,

Max Patiiuk, Grand Archmage of SDE II and Keeper of the Esri Chronicles 🧙‍♂️🪄🔮

---

lol, that was composed by GPT-4. The info it knows about me is from [custom instructions](https://openai.com/blog/custom-instructions-for-chatgpt) I have setup

**Comment by maxpatiiuk** [comment]
From @grantfitzsimmons:

> It would be nice to have a search function in the Query Builder to search the labels for all fields that are visible (and hidden if the show hidden items option is selected). This would speed up query building for users unfamiliar with the relational database components of Specify and want to find specific information.
> 
> **Requested By:** New Mexico Herbarium

Also from @grantfitzsimmons 😊:

> This is so smart
